code/test/security.js

```diff
@@ -126,22 +160,165 @@ contract('ONEWallet', (accounts) => {
+      const eotp = await ONEWallet.computeInnerEOTP({ otps })
+      // const treeIndex = Math.floor((NOW - effectiveTime) / INTERVAL) % 6
+      const treeIndex = c
+      Logger.debug({
+        otps: otps.map(e => {
+          const r = new DataView(new Uint8Array(e).buffer)
+          return r.getUint32(0, false)
+        }),
+        eotp: ONEUtil.hexString(eotp),
+        index,
+        treeIndex
+      })
+      Debugger.printLayers({ layers: innerTrees[treeIndex].layers })
+
+      const { tx, authParams, revealParams } = await TestUtil.commitReveal({
+        Debugger,
+        layers: innerTrees[treeIndex].layers,
+        index,
+        eotp,
+        paramsHash: ONEWallet.computeDataHash,
+        commitParams: { data: ONEUtil.hexStringToBytes(data) },
+        revealParams: { data, operationType: ONEConstants.OperationType.DISPLACE },
+        wallet
+      })
+      const successLog = tx.logs.find(log => log.event === 'CoreDisplaced') || tx.receipt.rawLogs.find(log => log.topics.includes('0x0b6dd4942da3070d72e5249990ad2b2703efd3f3a99c79cd0ce1a8eb50f8fdf4'))
+      if (checkDisplacementSuccess && !successLog) {
+        console.error(tx, authParams, revealParams)
+        Logger.debug(tx.receipt.rawLogs)
+        throw new Error('CoreDisplaced log missing')
+      }
+    }
+    return { wallet, newCore, newInnerCores, newKeys, newSeed: newComputedSeed, newEffectiveTime, newHseed, newLayers }
+    // assert.equal(ONE_CENT, balance, 'Wallet has correct balance')
+  }
+
+  const testForTime2 = async ({ multiple, effectiveTime, duration, seedBase = '0xdeadbeef1234567890023456789012', numTrees = 6, checkDisplacementSuccess = false, testTime = Date.now() }) => {
+    Logger.debug('testing:', { multiple, effectiveTime, duration })
+    const creationSeed = '0x' + (new BN(ONEUtil.hexStringToBytes(seedBase)).addn(duration).toString('hex'))
+    // const salt = 'SE-GENERAL-0-1' + Date.now()
+    let { walletInfo: alice, state } = await TestUtil.makeWallet({ salt: creationSeed, deployer: accounts[0], effectiveTime, duration })
+    // Start Tests
+    const newSeed = '0xdeedbeaf1234567890123456789012'
+    const testTimeNow = Math.floor(testTime / (INTERVAL)) * INTERVAL - 5000
+    const newEffectiveTime = Math.floor(testTimeNow / INTERVAL / 6) * INTERVAL * 6
+    const { core: newCore, innerCores: newInnerCores, identificationKeys: newKeys, vars: { seed: newComputedSeed, hseed: newHseed, client: { layers: newLayers } } } = await TestUtil.makeCores({
+      seed: newSeed,
+      effectiveTime: newEffectiveTime,
+      duration: duration,
+    })
+    const data = ONEWallet.encodeDisplaceDataHex({ core: newCore, innerCores: newInnerCores, identificationKey: newKeys[0] })
+    for (let c = 0; c < numTrees; c++) {
+      // Begin Tests
+      // alice changes the spending limit
+      let { tx, currentState } = await executeSecurityTransaction(
```

Comment:
you need to pass in the innerTrees and use the stuff there for layers. That's why this is failing but testForTime is not